src/app/DIAmeterApplication.cpp

```diff
@@ -73,241 +80,871 @@ int DIAmeterApplication::main(const vector<string>& input_files, const string in
+	  vector<InputFile> ms2_spectra_files = getInputFiles(input_files, 2);
+
+	  // Loop through spectrum files
+	  for (pair<vector<InputFile>::const_iterator, vector<InputFile>::const_iterator> f(ms1_spectra_files.begin(), ms2_spectra_files.begin());
+		 f.first != ms1_spectra_files.end() && f.second != ms2_spectra_files.end(); ++f.first, ++f.second) {
+
+		 string ms1_spectra_file = (f.first)->SpectrumRecords;
+		 string ms2_spectra_file = (f.second)->SpectrumRecords;
+		 string origin_file = (f.second)->OriginalName;
+
+		 // load MS1 and MS2 spectra
+
+		 // changing from loadMS1SpectraOld to loadMS1SpectraNew step-by-step
+		 // The main difference is to replace discretized mzbin with ppm-based
+		 // map<int, pair<double*, double*>> ms1scan_intensity_rank_map;
+		 // loadMS1SpectraOld(ms1_spectra_file, &ms1scan_intensity_rank_map);
+		 // carp(CARP_DEBUG, "old max_ms1scan:%d \t scan_gap:%d \t avg_noise_intensity_logrank:%f", max_ms1scan_, scan_gap_, avg_noise_intensity_logrank_);
+
+		 map<int, boost::tuple<double*, double*, double*, int>> ms1scan_mz_intensity_rank_map;
+		 map<int, boost::tuple<double, double>> ms1scan_slope_intercept_map;
+		 loadMS1SpectraNew(ms1_spectra_file, &ms1scan_mz_intensity_rank_map, &ms1scan_slope_intercept_map);
+		 carp(CARP_DEBUG, "new max_ms1scan:%d \t scan_gap:%d \t avg_noise_intensity_logrank:%f", max_ms1scan_, scan_gap_, avg_noise_intensity_logrank_);
+
+		 SpectrumCollection* spectra = loadSpectra(ms2_spectra_file);
+
+		 // insert the search code here and will split into a new function later
+		 double highest_ms2_mz = spectra->FindHighestMZ();
+		 MaxBin::SetGlobalMax(highest_ms2_mz);
+		 resetMods();
+		 carp(CARP_DEBUG, "Maximum observed MS2 m/z:%f", highest_ms2_mz);
+
+		 // Active queue to process the indexed peptides
+		 ActivePeptideQueue* active_peptide_queue = new ActivePeptideQueue(peptide_reader->Reader(), proteins);
+		 active_peptide_queue->setElutionWindow(0);
+		 active_peptide_queue->setPeptideCentric(false);
+		 active_peptide_queue->SetBinSize(bin_width_, bin_offset_);
+		 active_peptide_queue->SetOutputs(NULL, &locations, Params::GetInt("top-match"), true, output_file, NULL, highest_ms2_mz);
+
+		 // initialize fields required for output
+		 // Not sure if it is necessary, will check later
+		 const vector<SpectrumCollection::SpecCharge>* spec_charges = spectra->SpecCharges();
+		 int* sc_index = new int(-1);
+		 FLOAT_T sc_total = (FLOAT_T)spec_charges->size();
+		 int print_interval = Params::GetInt("print-search-progress");
+		 // Keep track of observed peaks that get filtered out in various ways.
+		 long int num_range_skipped = 0;
+		 long int num_precursors_skipped = 0;
+		 long int num_isotopes_skipped = 0;
+		 long int num_retained = 0;
+
+		 // This is the main search loop.
+		 // DIAmeter supports spectrum centric match report only
+		 if (Params::GetBool("peptide-centric-search")) { carp(CARP_FATAL, "Spectrum-centric match only!"); }
```

Comment:
Maybe move this if statement closer to the beginning of the file? This way there is less processing before it throws an error.